### PR TITLE
Fix for reordering of aerosols within modes.

### PR DESCRIPTION
**Pull request by overfelt**
Balwinder has a branch in which he changes the ordering of aerosols within modes. This should not make a difference but an extra call to aerosol_index_for_mode was breaking Gas/Aerosol exchange in this case.  Once this fix is merged into Balwinder's branch, it should fix his tests.

### CODE REVIEW

File: src/mam4xx/mo_setsox.hpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ struct Config {
   int itermax = 20;
   // Real small_value_20 = 1.0e-20;
   // Real small_value_30 = 1.0e-30;
-  int numptrcw_amode[AeroConfig::num_modes()] = {23, 28, 36, 40};
+  int numptrcw_amode[AeroConfig::num_modes()] = {22, 27, 35, 39};
```
Comment (singhbalwinder) [inline src/mam4xx/mo_setsox.hpp:78]: Is this change related to my species order change? To me, it looks like a fix to accommodate C++'s zero-based array indexing.

**Review by singhbalwinder — APPROVED** [review]
Thanks for fixing this!

File: src/mam4xx/mo_setsox.hpp (hunk 1 above)
Comment (overfelt) [reply src/mam4xx/mo_setsox.hpp:78]: Yes, it has nothing to do with your species order change and is just a Fortran to C++ indexing problem. But the commit integration tests failed due to the memory error and i needed to fix it to pass all of the tests.
